challengeMk2/ViewModels/SystemDetailCarouselViewModel.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using challengeMk2.Models;
+using Newtonsoft.Json;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+
+namespace challengeMk2.ViewModels
+{
+    public class SystemDetailCarouselViewModel : BaseViewModel
+    {
+        private StarSystem currentSystem;
+        public StarSystem CurrentSystem
+        {
+            get => currentSystem;
+            set
+            {
+                SetProperty<StarSystem>(ref currentSystem, value);
+            }
+        }
+
+        private StarSystem detailedSystem;
+        public StarSystem DetailedSystem
+        {
+            get => detailedSystem;
+            set
+            {
+                SetProperty<StarSystem>(ref detailedSystem, value);
+            }
+        }
+
+
+        private int currentBodyCount;
+        public int CurrentBodyCount
+        {
+            get => currentBodyCount;
+            set
+            {
+                SetProperty<int>(ref currentBodyCount, value);
+            }
+        }
+
+        private double currentDistance;
+        public double CurrentDistance
+        {
+            get => currentDistance;
+            set
+            {
+                SetProperty<double>(ref currentDistance, value);
+            }
+        }
+
+
+        public NetworkAccess CurrentConnectivity { get; set; }
+
+        public IList SystemInfos { get; set; }
+
+
+
+
+
+
+
+
+        public SystemDetailCarouselViewModel(StarSystem selectedSystem = null)
+        {
+            Title = "Star System Details";
+
+            SystemInfos = new ObservableCollection<StarSystem>();
+
+            currentSystem = selectedSystem;
+
+            //Get infos
+            UpdateSystemData();
+        }
+
+
+
+
+
+
+
+
+        private async void UpdateSystemData()
+        {
+            CurrentConnectivity = Connectivity.NetworkAccess;
+
+            //1° Get details from online API
+            if (CurrentConnectivity == NetworkAccess.Internet)
+            {
+                DetailedSystem = await GetDetailsFromApi();
+            }
+            else
+            {
+                DetailedSystem.Name = "No internet connection ! Try again later...";
+                return;
+            }
+
+            //2° Get distance and number of bodies from currentSystem and put them in detailedSystem
+            GetCompInfos();
+
+            //3° Fill collection for Carousel : Copy 3 times detailed system in SystemInfos (one for each tab)
+            FillSystemInfos();
+        }
+
+        private async Task<StarSystem> GetDetailsFromApi()
+        {
+            HttpClient client = new HttpClient();
```

Comment:
Don't forget to dispose the HttpClient. You could use the C# 8.0 using declaration syntax.

Reply:
I came across this article about HttpClient : https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/
What do you thing is better, use using or a static HttpClient ?